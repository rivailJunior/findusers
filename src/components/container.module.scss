.mainContainer {
    // position: absolute;
    display: grid;
    height: auto;

    // background-color: #eaeaead6;
    grid-template-columns: repeat(12, 1fr);
    align-self: center;
    margin-right: 15%;
    margin-left: 15%;
}

@mixin grid-column($columnSize) {
    grid-column: span $columnSize;
    min-height: 50px;
    padding: 10px;
}

$list: 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12;

@each $size in $list {
    .containerItem#{$size} {
        @include grid-column($size);
    }
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    .mainContainer {
        margin-right: 1%;
        margin-left: 1%;
    }
    @each $size in $list {
        .containerItem#{$size} {
            @include grid-column(12);
        }
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width: 481px) and (max-device-width: 1024px) {
}
