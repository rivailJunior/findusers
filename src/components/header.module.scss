.header {
    overflow: hidden;
    background-color: white;
    padding: 10px 10px;
}

.header a {
    float: left;
    color: black;
    text-align: center;
    padding: 12px;
    text-decoration: none;
    font-size: 18px;
    line-height: 25px;
    border-radius: 4px;
}

.header a.logo {
    font-size: 25px;
    font-weight: bold;
}

.header a:hover {
    background-color: #ddd;
    color: black;
    cursor: pointer;
}

.header a.active {
    background-color: dodgerblue;
    color: white;
}

.header-right {
    float: right;
}

@media screen and (max-width: 500px) {
    .header a {
        float: none;
        display: block;
        text-align: left;
    }

    .header-right {
        float: none;
    }
}
