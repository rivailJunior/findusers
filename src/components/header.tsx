import React from 'react';
import styles from './header.module.scss'
import { SearchInput } from './searchInput';

type iHeader = {
    userName?: string,
    labelRight?: string,
    handleRight?(): Function
    handleSearch?: Function
}

const Header = ({ userName, labelRight, handleRight, handleSearch }: iHeader): JSX.Element => {
    return (
        <div className={styles.header}>
            <div className={styles.headerLeft}>
                <a href="#default" className={styles.logo}>{userName}</a>
            </div>
            <div className={styles.headerRight}>
                <div>
                    <SearchInput handleSearch={handleSearch} placeholder="Buscar Usuario" />
                    {labelRight && (
                        <div className={styles['header-right']}>
                            <a className={styles.active} onClick={handleRight} >Sair</a>
                        </div>
                    )}
                </div>

            </div>
        </div>

    );
}

export { Header };
