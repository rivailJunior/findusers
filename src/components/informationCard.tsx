import React from 'react';
import styles from './informationCard.module.scss';

type iCard = {
    title: string,
    mainText: string,

}
const InformationCard = ({ title, mainText }: iCard) => {
    return (
        <div className={styles.card}>
            <p>{title}</p>
            <h2>{mainText}</h2>
        </div>
    );
}

export { InformationCard };
