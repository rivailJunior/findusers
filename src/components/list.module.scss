@use '../pages/index.scss' as index;
@use '../pages/root.scss' as root;
.item {
    background-color: white;
    border: 0.5px solid #9e9e9e80;
    display: flex;
    flex-direction: column;
    overflow: hidden;
    border-radius: 5px;

    .itemLi {
        @include index.default-label(15px, 0, 100%, 300, 15px);
        display: grid;
        text-align: left;
        padding: 1%;
        display: grid;
        grid-template-columns: repeat(12, 1fr);
        max-height: 100px;

        .left {
            grid-column: span 2;
            display: flex;
            justify-content: center;
            align-self: center;
            font-size: 3rem;
        }
        .right {
            grid-column: span 10;

            .rightTitle {
                @include index.default-label(15px, 0, 100%, 700, 20px);
                // font-size: 1rem;
                text-transform: capitalize;
                text-align: left;
            }
            .rightSub {
                @include index.default-label(15px, 1%, 100%, 300, 20px, left);
                display: flex;
                flex-direction: column;
                justify-content: space-between;
            }
        }
    }
    .itemLi:hover {
        cursor: pointer;
        background-color: root.$hoverBgc;
        color: root.$hoverColor;
    }
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    .item {
        .itemLi {
            @include index.default-label(12px, 0, 100%, 500);

            grid-template-columns: repeat(12, 1fr);

            .left {
                display: none;
            }
            .right {
                grid-column: span 12;

                .rightTitle {
                    font-size: 1rem;
                    text-transform: capitalize;
                }
                .rightSub {
                    @include index.default-label(
                        10px,
                        1%,
                        100%,
                        300,
                        10px,
                        left
                    );
                    display: flex;
                    flex-direction: column;
                    justify-content: space-between;
                }
            }
        }
        .itemLi:hover {
            cursor: pointer;
            background-color: root.$hoverBgc;
            color: root.$hoverColor;
        }
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width: 481px) and (max-device-width: 1024px) {
}

/* Desktops and laptops ----------- */
@media only screen and (min-width: 1224px) {
    /* Styles */
}
