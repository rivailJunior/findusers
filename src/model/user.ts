export const UserValue: User = {
    avatar_url: "https://avatars.githubusercontent.com/u/5783143?v=4",
    bio: "Full Stack developer, javascript lover and an open source enthusiastic.",
    blog: "",
    company: "Qtarkuss",
    created_at: "2013-10-26T19:17:03Z",
    email: "rivail.rj@gmail.com",
    events_url: "https://api.github.com/users/rivailJunior/events{/privacy}",
    followers: 4,
    followers_url: "https://api.github.com/users/rivailJunior/followers",
    following: 2,
    following_url: "https://api.github.com/users/rivailJunior/following{/other_user}",
    gists_url: "https://api.github.com/users/rivailJunior/gists{/gist_id}",
    gravatar_id: "",
    hireable: true,
    html_url: "https://github.com/rivailJunior",
    id: 0,
    location: "Brasil",
    login: "rivailJunior",
    name: "Rivail Santos",
    node_id: "MDQ6VXNlcjU3ODMxNDM=",
    organizations_url: "https://api.github.com/users/rivailJunior/orgs",
    public_gists: 0,
    public_repos: 12,
    received_events_url: "https://api.github.com/users/rivailJunior/received_events",
    repos_url: "https://api.github.com/users/rivailJunior/repos",
    site_admin: false,
    starred_url: "https://api.github.com/users/rivailJunior/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/rivailJunior/subscriptions",
    twitter_username: null,
    type: "User",
    updated_at: "2021-04-25T03:20:04Z",
    url: "https://api.github.com/users/rivailJunior"
}

export type User = {
    avatar_url: string
    bio: string
    blog: string
    company: string
    created_at: string
    email: string
    events_url: string
    followers: number
    followers_url: string
    following: number
    following_url: string
    gists_url: string
    gravatar_id: string
    hireable: boolean
    html_url: string
    id: number
    location: string
    login: string
    name: string
    node_id: string
    organizations_url: string
    public_gists: number
    public_repos: number
    received_events_url: string
    repos_url: string
    site_admin: false
    starred_url: string
    subscriptions_url: string
    twitter_username: null
    type: string
    updated_at: string
    url: string
}