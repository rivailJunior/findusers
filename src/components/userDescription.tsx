import React from 'react';
import { User } from '../model/user';
import styles from './userDescription.module.scss'

type iDescription = {
    userData: User
}
const UserDescription = ({ userData }: iDescription): JSX.Element => {
    return (
        <div className={styles.card}>
            <div className={styles.left}>
                {userData?.avatar_url ? (
                    <img src={userData?.avatar_url} alt="Imagem do usuÃ¡rio" />
                ) : (
                    <img src={'/noimg.jpg'} alt="Imagem nao encontrada" />
                )}

            </div>
            <div className={styles.right}>
                <div className={styles.title}>{userData?.email}</div>
                <div className={styles.subTitle}>{userData?.bio}</div>
                <div className={styles.subTitle}>Seguidores: {userData?.followers}</div>
                <div className={styles.subTitle}>Seguindo: {userData?.following}</div>
            </div>
        </div>
    );
}

export { UserDescription };