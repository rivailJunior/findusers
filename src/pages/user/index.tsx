import React, { useEffect } from 'react'
import { GetServerSideProps } from 'next'
import { doPostRequestLogin } from '../../provider/requestProvider'
import { handleRedirect, handleSession } from '../../utils/sessionValidate'
import useSession from '../../context/useSession'
import useRepositories from '../../context/useRepositories'
import { ListRepositories } from '../../components/list'
import { GridContainer, GridContainerItem } from '../../components/gridContainer'
import { ErrorFeedback } from '../../components/errorFeedback'
import Header from '../../components/header'
import { UserDescription } from '../../components/userDescription'
import { useRouter } from 'next/router'
import BreadCrumb from '../../components/breadCrumb';
import { breadCrumbTitles } from '../../utils/breadCrumbTitles'


const Index = ({ userData, token }): JSX.Element => {
	const router = useRouter()
	const { user, removeSession, findUser, findError } = useSession(token, userData)
	const { findRepositories, repositories } = useRepositories();

	useEffect(() => {
		//here will find after login
		if (userData) {
			findRepositories(userData?.login)
		}
	}, [])

	useEffect(() => {
		// here will change when search a new user
		if (user) {
			findRepositories(user?.login)
		}
	}, [user])

	const searchUser = async (userName) => {
		await findUser(userName)
	}

	const handleOpenRepository = (repository) => {
		router.push({
			pathname: '/repositories',
			query: {
				repository: repository.name
			}
		})
	}

	return (
		<div>
			<Header handleSearch={searchUser} userName={user?.name} labelRight="Sair" handleRight={() => removeSession()} />
			{!findError ? (
				<>
					<GridContainer>
						<GridContainerItem size={12}>
							<BreadCrumb titles={breadCrumbTitles} />
						</GridContainerItem>
						<GridContainerItem size={12}>
							<UserDescription userData={user} />
						</GridContainerItem>
					</GridContainer>
					<GridContainer>
						<GridContainerItem size={12}>
							<ListRepositories data={repositories} handleClick={handleOpenRepository} />
						</GridContainerItem>
					</GridContainer>
				</>
			) : (
				<ErrorFeedback title="Nenhum novo usuario encontrado" />
			)}

		</div>

	)
}

export const getServerSideProps: GetServerSideProps = async ({
	req,
	res,
	query
}) => {
	try {
		const code = query.code;
		const accessToken = handleSession(req, res)
		const { user, token } = await doPostRequestLogin(code, accessToken)
		return {
			props: {
				userData: user,
				token: token
			}
		}
	} catch (err) {
		if (err === 'bad_verification_code' || err?.response?.status === 401) {
			handleRedirect(req, res)
		}
		return {
			props: {}
		}
	}
}
export default Index
