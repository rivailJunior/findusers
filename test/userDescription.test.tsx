import React from 'react'
import { unmountComponentAtNode } from "react-dom";
import { render, cleanup, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { UserDescription } from '../src/components/userDescription'
import { UserValue } from '../src/model/user'

let container = null;
beforeEach(() => {
    // configurar o elemento do DOM como o alvo da renderização
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // limpar na saída
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe('User Description Component', () => {
    test('Render Correctly', () => {
        const container = renderer.create(<UserDescription userData={UserValue} />, document)
        container.toJSON();
        expect(container).toMatchSnapshot()
    })

    test('Show all informations descriptions on Component', () => {
        render(<UserDescription userData={UserValue} />, document)
        expect(screen.getByText(/rivail.rj@gmail.com/i)).toBeInTheDocument()
        expect(screen.getByText(/Seguidores: /i)).toBeInTheDocument()
        expect(screen.getByText(/Seguindo: /i)).toBeInTheDocument()
        expect(screen.queryByAltText(/Imagem do usuário/i)).toBeInTheDocument()
        expect(screen.getByText(/Full Stack developer, javascript lover and an open source enthusiastic./i)).toBeInTheDocument()
    })

    test('Render with user img', () => {
        render(<UserDescription userData={UserValue} />, document)
        expect(screen.queryByAltText(/Imagem do usuário/i)).toBeInTheDocument()
        expect(screen.queryByAltText(/Imagem nao encontrada/i)).not.toBeInTheDocument()
    })

    test('Render without user img', () => {
        UserValue.avatar_url = ""
        render(<UserDescription userData={UserValue} />, document)
        expect(screen.queryByAltText(/Imagem do usuário/i)).not.toBeInTheDocument()
        expect(screen.queryByAltText(/Imagem nao encontrada/i)).toBeInTheDocument()
    })

})