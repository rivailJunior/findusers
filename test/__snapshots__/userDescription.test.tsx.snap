// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Description Component Render Correctly 1`] = `
<div
  className="card"
>
  <div
    className="left"
  >
    <img
      alt="Imagem do usuÃ¡rio"
      src="https://avatars.githubusercontent.com/u/5783143?v=4"
    />
  </div>
  <div
    className="right"
  >
    <div
      className="title"
    >
      rivail.rj@gmail.com
    </div>
    <div
      className="subTitle"
    >
      Full Stack developer, javascript lover and an open source enthusiastic.
    </div>
    <div
      className="subTitle"
    >
      Seguidores: 
      4
    </div>
    <div
      className="subTitle"
    >
      Seguindo: 
      2
    </div>
  </div>
</div>
`;
